hostmap.go

```diff
@@ -245,9 +348,37 @@ func (hm *HostMap) DeleteReverseIndex(index uint32) {
 }
 
 func (hm *HostMap) DeleteHostInfo(hostinfo *HostInfo) {
```

Comment:
I mostly punted on thinking through locking here. Looking at it now. I'm trying to come up with a locking strategy that avoids the possibility of deadlocks and is correct, but it's a bit challenging.

Suggestions here are welcome!

Reply:
I'm going to try sticking all the relay variables into a struct that has its own lock, so they can be managed independently from the HostInfo lock, but still be safe for concurrent access.